src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -386,6 +397,22 @@ public void validate(FailureCollector collector) {
       }
     }
 
+    //This method validated the pattern of CMEK Key resource ID.
+    private void validateCmekKey(FailureCollector failureCollector) {
+      String[] part = cmekKey.split("/");
+      if ((part.length != 8) || !part[0].equals("projects") || !part[2].equals("locations")
+        || !part[4].equals("keyRings") || !part[6].equals("cryptoKeys")) {
+        failureCollector.addFailure("Invalid cmek key resource Id format. It should be of the form " +
+                                      "projects/<gcp-project-id>/locations/<key-location>/keyRings/" +
+                                      "<key-ring-name>/cryptoKeys/<key-name>", null)
+          .withConfigProperty(NAME_CMEK_KEY);
+      } else if (!part[3].equals(location)) {
+        failureCollector.addFailure(String.format("Cloud KMS region '%s' not available for use " +
```

Comment:
The plugin behavior (at least according to the description) is that the location is ignored if the bucket already exists.  This should only be checked if the bucket does not exist.

The message should also be more specific. For example, the message can be "CMEK key '%s' is in location '%s', while the GCS bucket will be created in location '%s'."  and the corrective action can be "Modify the CMEK key or bucket location to be the same"